src/Expression/Simplifier.purs

```diff
@@ -0,0 +1,70 @@
+simplify (ExpressionUnary operation expression) = case simplify expression, operation of
+  ExpressionLiteral value, Neg -> ExpressionLiteral (-value)
+  simplifiedExpression, _ -> ExpressionUnary operation simplifiedExpression
+
+simplify (ExpressionBinary Power leftExpression (ExpressionLiteral 1.0)) = simplify leftExpression
+
+simplify (ExpressionBinary Power _ (ExpressionLiteral 0.0)) = ExpressionLiteral 1.0
+
+simplify (ExpressionBinary Power (ExpressionVariable "e") rightExpression) = ExpressionUnary Exp $ simplify rightExpression
+
+simplify (ExpressionBinary operation leftExpression rightExpression) =
+  fromMaybe (ExpressionBinary operation simplifiedLeftExpression simplifiedRightExpression)
+    $ trimSimpleNodes simplifiedLeftExpression simplifiedRightExpression operation
+    <|> trimConstantLeafNodes simplifiedLeftExpression simplifiedRightExpression operation
+    <|> trimTimesOperations simplifiedLeftExpression simplifiedRightExpression operation
+    <|> trimPlusOperations simplifiedLeftExpression simplifiedRightExpression operation
+  where
+  simplifiedLeftExpression = simplify leftExpression
+
+  simplifiedRightExpression = simplify rightExpression
+
+simplify expression = expression
+
+trimSimpleNodes :: Expression -> Expression -> BinaryOperation -> Maybe Expression
+trimSimpleNodes simplifiedLeftExpression simplifiedRightExpression operation = case simplifiedLeftExpression, simplifiedRightExpression, operation of
+  ExpressionLiteral 0.0, _, Times -> Just $ ExpressionLiteral 0.0
+  _, ExpressionLiteral 0.0, Times -> Just $ ExpressionLiteral 0.0
+  ExpressionLiteral 1.0, _, Times -> Just $ simplifiedRightExpression
+  _, ExpressionLiteral 1.0, Times -> Just $ simplifiedLeftExpression
+  ExpressionLiteral 0.0, _, Plus -> Just $ simplifiedRightExpression
+  _, ExpressionLiteral 0.0, Plus -> Just $ simplifiedLeftExpression
+  ExpressionLiteral 0.0, _, Minus -> Just $ ExpressionUnary Neg simplifiedRightExpression
+  _, ExpressionLiteral 0.0, Minus -> Just $ simplifiedLeftExpression
+  ExpressionLiteral 0.0, _, Divide -> Just $ ExpressionLiteral 0.0
+  _, _, _ -> Nothing
+
+trimConstantLeafNodes :: Expression -> Expression -> BinaryOperation -> Maybe Expression
+trimConstantLeafNodes simplifiedLeftExpression simplifiedRightExpression operation = case simplifiedLeftExpression, simplifiedRightExpression, operation of
+  ExpressionLiteral leftValue, ExpressionLiteral rightValue, Plus -> Just $ ExpressionLiteral (leftValue + rightValue)
+  ExpressionLiteral leftValue, ExpressionLiteral rightValue, Minus -> Just $ ExpressionLiteral (leftValue - rightValue)
+  ExpressionLiteral leftValue, ExpressionLiteral rightValue, Times -> Just $ ExpressionLiteral (leftValue * rightValue)
+  ExpressionLiteral leftValue, ExpressionLiteral rightValue, Divide -> Just $ ExpressionLiteral (leftValue / rightValue)
+  ExpressionLiteral leftValue, ExpressionLiteral rightValue, Power -> Just $ ExpressionLiteral (pow leftValue rightValue)
```

Comment:
OK for now, but when we switch to Rational numbers, this simplification will be possible only when the result is rational.

Reply:
Ok noted